### PR TITLE
`weather-dl-v2` improvements 5

### PR DESCRIPTION
**Pull request by aniketinfocusp**
This PR contains:
- marking status of licenses
- license redeploy command for CLI

### CODE REVIEW

File: weather_dl_v2/cli/app/subcommands/license.py [hunk 1]
```diff
@@ -102,3 +102,30 @@ def update_license(
         return
 
     print(license_service._update_license(license, license_dict))
+
+@app.command("redeploy",
+             help="""Redeploy licenses."""
+             """\nCAUTION: Redeploying will cause licenses to stop whatever they are doing."""
+             """This can cause queues to be filled with stray reqs from previous deployments."""
```
Comment (mahrsee1997) [inline weather_dl_v2/cli/app/subcommands/license.py:109]: nit: requests.

File: weather_dl_v2/cli/app/subcommands/license.py [hunk 2]
```diff
@@ -102,3 +102,30 @@ def update_license(
...
+            )
+def redeploy_license(
+    license_id: Annotated[
+        str,
+        typer.Option(
+            help="""Mention license_id of license to redeploy"""
```
Comment (mahrsee1997) [inline weather_dl_v2/cli/app/subcommands/license.py:115]: nit: `.`

File: weather_dl_v2/cli/app/subcommands/license.py (hunk 1 above)
Comment (aniketinfocusp) [reply weather_dl_v2/cli/app/subcommands/license.py]: done

File: weather_dl_v2/cli/app/subcommands/license.py (hunk 2 above)
Comment (aniketinfocusp) [reply weather_dl_v2/cli/app/subcommands/license.py]: done

File: weather_dl_v2/cli/app/services/license_service.py [hunk 3]
```diff
@@ -48,6 +48,13 @@ def _remove_license(self, license_id: str):
     def _update_license(self, license_id: str, license_dict: dict):
         pass
 
+    @abc.abstractmethod
+    def _redeploy_license_by_license_id(self, license_id: str):
+        pass
+
+    @abc.abstractmethod
+    def _redeploy_licenses_by_client(self, license_id: str):
```
Comment (mahrsee1997) [inline weather_dl_v2/cli/app/services/license_service.py:56]: `client_name`.

File: weather_dl_v2/license_deployment/database.py [hunk 4]
```diff
@@ -143,6 +143,16 @@ def _empty_license_queue(self, license_id: str) -> None:
             f"Updated {license_id} queue in 'queues' collection. Update_time: {result.update_time}."
         )
 
+    def _mark_license_status(self, license_id: str, status: str) -> None:
```
Comment (mahrsee1997) [inline weather_dl_v2/license_deployment/database.py:146]: Add abstract method in CRUDOperations.

File: weather_dl_v2/fastapi-server/routers/download.py [hunk 5]
```diff
@@ -213,6 +214,7 @@ async def submit_download(
             )
 
         for license_id in licenses:
+            await mark_license_active(license_id, license_handler)
```
Comment (mahrsee1997) [inline weather_dl_v2/fastapi-server/routers/download.py:217]: Lets move this below the `if` block.

File: weather_dl_v2/fastapi-server/routers/download.py [hunk 6]
```diff
@@ -375,6 +377,7 @@ async def retry_config(
         )
 
     for license_id in licenses:
+        await mark_license_active(license_id, license_handler)
```
Comment (mahrsee1997) [inline weather_dl_v2/fastapi-server/routers/download.py:380]: Check above comment.

File: weather_dl_v2/fastapi-server/routers/download.py (hunk 5 above)
Comment (aniketinfocusp) [reply weather_dl_v2/fastapi-server/routers/download.py]: done in 
82e3cf4

File: weather_dl_v2/fastapi-server/routers/download.py (hunk 6 above)
Comment (aniketinfocusp) [reply weather_dl_v2/fastapi-server/routers/download.py]: done in 
82e3cf4

File: weather_dl_v2/cli/app/services/license_service.py (hunk 3 above)
Comment (aniketinfocusp) [reply weather_dl_v2/cli/app/services/license_service.py]: fixed in [e892514](https://github.com/google/weather-tools/pull/420/commits/e892514e70b7ed5bc1509848f9d36eca289ec872)

**Review by mahrsee1997 — APPROVED** [review]
LGTM.

**Pull request merged**